src/Microsoft.DotNet.Darc/src/Darc/Operations/AddBuildToChannelOperation.cs

```diff
@@ -178,6 +187,46 @@ private async Task<int> PromoteBuildAsync(Build build, Channel targetChannel, IR
             }
         }
 
+        private async Task<bool> ValidateAzDOBuildAsync(AzureDevOpsClient azdoClient, string azureDevOpsAccount, string azureDevOpsProject, int azureDevOpsBuildId)
+        {
+            try
+            {
+                var artifacts = await azdoClient.GetBuildArtifactsAsync(azureDevOpsAccount, azureDevOpsProject, azureDevOpsBuildId, maxRetries: 5);
+
+                // The build manifest is always necessary
+                if (!artifacts.Any(f => f.Name.Equals("AssetManifests")))
+                {
+                    Console.Write("The build that you want to promote doesn't have a Build Manifest. That's required for publishing. Aborting.");
+                    return true;
+                }
+
+                if ((_options.DoSigningValidation || _options.DoNuGetValidation || _options.DoSourcelinkValidation)
+                    && !artifacts.Any(f => f.Name.Equals("PackageArtifacts")))
+                {
+                    Console.Write("The build that you want to promote doesn't have a list of package assets. That's required when running signing or NuGet validation. Aborting.");
+                    return true;
+                }
+
+                if (_options.DoSourcelinkValidation && !artifacts.Any(f => f.Name.Equals("BlobArtifacts")))
+                {
+                    Console.Write("The build that you want to promote doesn't have a list of blob assets. That's required when running SourceLink validation. Aborting.");
+                    return true;
+                }
+
+                return true;
+            }
+            catch (HttpRequestException e) when (e.Message.Contains("404 (Not Found)"))
+            {
+                Console.Write("The build that you want to promote isn't available in AzDO anymore. Aborting.");
+                return false;
+            }
+            catch (HttpRequestException e) when (e.Message.Contains("401 (Unauthorized)"))
+            {
+                Console.Write("Got permission denied response while trying to retrieve target build from Azure DevOps. Aborting.");
```

Comment:
Think we can be a little bit more helpful here:
Please make sure that you have access to the build definition that is being added to the new channel, and that your Azure DevOps PAT has the build read and execute scopes set?

Reply:
The PAT needs permission to the promotion build definition AFAIU. I changed the text to this "Please make sure that your Azure DevOps PAT has the build read and execute scopes set.".